packages/wonder-blocks-popover/src/components/__tests__/focus-manager.test.tsx

```diff
@@ -109,4 +87,105 @@ describe("FocusManager", () => {
+            name: "Open popover",
+        });
+
+        render(
+            <FocusManager anchorElement={anchorElementNode}>
+                <div>
+                    <button>first focusable element inside</button>
+                </div>
+            </FocusManager>,
+        );
+
+        // Act
+        // 1. focus on the previous element before the popover
+        userEvent.tab();
+
+        // 2. focus on the anchor element
+        userEvent.tab();
+
+        // 3. focus on focusable element inside the popover
+        userEvent.tab();
+
+        // 4. focus on the next focusable element outside the popover (this will
+        //    be the first focusable element outside the popover)
+        userEvent.tab();
+
+        // NOTE: At this point, the focus moves to the document body, so we need
+        // to press tab again to move the focus to the next focusable element.
+        userEvent.tab();
+
+        // 5. Finally focus on the first element in the document
+        userEvent.tab();
+
+        // find previous focusable element outside the popover
+        const prevFocusableElementOutside = screen.getByRole("button", {
+            name: "Prev focusable element outside",
+        });
+
+        // Assert
+        expect(prevFocusableElementOutside).toHaveFocus();
+    });
+
+    it("should disallow focusability on internal elements if the user focus out of the focus manager", async () => {
+        // Arrange
+        const externalNodes = (
+            <div>
+                <button>Prev focusable element outside</button>
+                <button>Open popover</button>
+                <button>Next focusable element outside</button>
+            </div>
+        );
+        render(externalNodes);
+
+        // get the anchor reference to be able pass it to the FocusManager
+        const anchorElementNode = screen.getByRole("button", {
+            name: "Open popover",
+        });
+
+        render(
+            <FocusManager anchorElement={anchorElementNode}>
+                <div>
+                    <button>first focusable element inside</button>
+                </div>
+            </FocusManager>,
+        );
+
+        // Act
+        // 1. focus on the previous element before the popover
+        userEvent.tab();
+
+        // 2. focus on the anchor element
+        userEvent.tab();
+
+        // 3. focus on focusable element inside the popover
+        userEvent.tab();
+
+        // 4. focus on the next focusable element outside the popover (this will
+        //    be the first focusable element outside the popover)
+        userEvent.tab();
+
+        // The elements inside the focus manager should not be focusable anymore.
+        const focusableElementInside = screen.getByRole("button", {
+            name: "first focusable element inside",
+        });
+
+        // Assert
+        expect(focusableElementInside).toHaveAttribute("tabIndex", "-1");
+    });
```

Comment:
**praise:** thanks for labelling what each `userEvent` method call is for.

Reply:
Does tabbing back into the popover making elements inside the popover focusable again?

Reply:
That's correct.... I make them focusable as soon as the focus goes back to the popover wrapper. One important thing to mention is that I just noticed an issue when the trigger element is the last focusable element in the DOM. I might need to completely change this approach if needed, so wanted to give a heads up.